Share with Gifski/ShareViewController.swift

```diff
@@ -0,0 +1,53 @@
+import Cocoa
+
+final class ShareViewController: NSViewController {
+	override var nibName: NSNib.Name? { NSNib.Name("ShareViewController") }
+
+	override func loadView() {
+		super.loadView()
+
+		guard let item = (self.extensionContext?.inputItems[0] as? NSExtensionItem)?.attachments?.first else {
+			errorOpenMainApp()
+			return
+		}
+
+		var typeIdentifier: String
+		if item.hasItemConformingToTypeIdentifier("public.mpeg-4") {
+			typeIdentifier = "public.mpeg-4"
+		} else if item.hasItemConformingToTypeIdentifier("com.apple.m4v-video") {
+			typeIdentifier = "com.apple.m4v-video"
+		} else if item.hasItemConformingToTypeIdentifier("com.apple.quicktime-movie") {
+			typeIdentifier = "com.apple.quicktime-movie"
+		} else {
+			errorOpenMainApp()
+			return
+		}
+
+		item.loadFileRepresentation(forTypeIdentifier: typeIdentifier) { url, _ in
+			guard let url = url else {
+				self.errorOpenMainApp()
+				return
+			}
+
+			let shareUrl = "\(url.lastPathComponent)"
+			let appIdentifierPrefix = Bundle.main.infoDictionary!["AppIdentifierPrefix"] as! String
+			let appGroupShareVideUrl = FileManager.default.containerURL(forSecurityApplicationGroupIdentifier: "\(appIdentifierPrefix).gifski_video_share_group")?.appendingPathComponent(shareUrl)
+
+			try? FileManager.default.removeItem(at: appGroupShareVideUrl!)
+			try! FileManager.default.copyItem(at: url, to: appGroupShareVideUrl!)
```

Comment:
Too much force-unwrapping. You can not be sure all of this is going to work.

Reply:
Actually it's well documented in apples libraries. `appGroupShareVideUrl ` can never be nil as it will never change. `FileManager.default.copyItem ` will only throw if either the destination file exists, which is impossible because of the line before, or you don't have permissions, which is impossible in an app group.

But I changed it to handle this case.